src/alveoleye/_export_operations.py

```diff
@@ -1,136 +1,152 @@
+
+    for idx, r in enumerate(results, start=1):
+        d = r.to_dict()
+        d["case_id"] = idx
+        rows.append(d)
+
+    if fmt.lower() == "csv":
+        fieldnames = ["case_id"] + [k for k in rows[0] if k != "case_id"] if rows else []
+        with open(out_path, "w", newline="") as fh:
+            writer = csv.DictWriter(fh, fieldnames=fieldnames)
+            writer.writeheader()
+            writer.writerows(rows)
+    else:
+        payload = {str(idx): {**r.to_dict(), "case_id": idx}
+                   for idx, r in enumerate(results, start=1)}
+        with open(out_path, "w") as fh:
+            json.dump(payload, fh, indent=2)
+
 
-    return file_name
+def write_labelmaps(results: List[Result], labelmap_dir: str, ext: str):
+    for idx, r in enumerate(results, start=1):
+        if not r.labelmaps:
+            continue
 
+        result_dir = os.path.join(labelmap_dir, str(idx))
+        os.makedirs(result_dir, exist_ok=True)
 
-def export_accumulated_results(accumulated_results, output_dir, file_name="test_results.csv"):
-    if not output_dir:
-        return
+        for layer_name, lm in r.labelmaps.items():
+            fn = f"{layer_name}.{ext}"
+            outp = os.path.join(result_dir, fn)
+            tifffile.imwrite(outp, lm.astype("uint16"))
 
-    csv_data = create_csv_data(accumulated_results)
 
-    os.makedirs(output_dir, exist_ok=True)
-    unique_file_name = get_unique_filename(output_dir, file_name)
-    complete_export_path = os.path.join(output_dir, unique_file_name)
+def write_images(results: List[Result], labelmap_dir: str, ext: str):
+    colormap = _norm_to_rgb(Config.get_label_indexed_colormap())
 
-    with open(complete_export_path, "w", newline="") as results_file:
-        results_file.write(csv_data)
+    for idx, r in enumerate(results, start=1):
+        if not r.labelmaps:
+            continue
+
+        result_dir = os.path.join(labelmap_dir, str(idx))
+        os.makedirs(result_dir, exist_ok=True)
+
+        for layer_name, arr in r.labelmaps.items():
+            arr = np.asarray(arr)
+
+            if arr.ndim == 3 and arr.shape[2] == 3:
+                rgb_image = arr.astype(np.uint8)
+
+            else:
+                if arr.ndim == 3 and arr.shape[0] == 1:
+                    lm = arr[0]
+                elif arr.ndim == 2:
+                    lm = arr
+                else:
+                    print(f"[!] Skipping {layer_name!r}: unsupported shape {arr.shape}")
+                    continue
+
+                h, w = lm.shape
+                rgb_image = np.zeros((h, w, 3), dtype=np.uint8)
+                for label, color in colormap.items():
+                    rgb_image[lm == label] = color
+
+            fn = f"{layer_name}.{ext}"
+            outp = os.path.join(result_dir, fn)
+            Image.fromarray(rgb_image).save(outp)
+            
+
+def zip_folder(src_folder: str, zip_target: str):
+    root, folder = os.path.split(src_folder.rstrip("/\\"))
```

Comment:
When you have time, could you write a test for this? I haven't followed the entire code path to check, but different operating systems might have different slashes here, making the rstrip not work as expected.